### PR TITLE
thin_provisioning cases using local images and gluster images

### PR DESCRIPTION
**Pull request by PyLearner**
ID: 1352781, 1352782
Signed-off-by: Meng Yang meyang@redhat.com

this patch is mainly related to use local image and gluster image to do some test.

### CODE REVIEW

**Comment by suqinhuang** [comment]
id?

**Comment by PyLearner** [comment]
ID: 1352781, 1352782

File: qemu/tests/thin_provisioning_image.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      :param env: Dictionary with test environment.
+      """
+
+    def count_block_num(image_disk, gluster_mount_dir=None):
+        """
+        Get the block num of the disk,when i create a large file in guest,the content needs
+        time to be written to the host disk,so when check using stat, you will find the block
+        number changes all the time for a period of time, this is related with the host env,
+        some machines is much faster,and some machines is slow, i test 3 host machines and finally
+        find when i set the wait_time to 12 seconds,it works fine..
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: can you short the description?

File: qemu/tests/thin_provisioning_image.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+        :param image_disk: Image disk name.
+        :return: The block number after the status is steady
+        """
+        wait_time = 20
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: it's better to set it configurable

File: qemu/tests/thin_provisioning_image.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        while True:
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: why need while ?

File: qemu/tests/thin_provisioning_image.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            if params.get("enable_gluster", "no") == "yes":
+                image_disk = process.system_output("basename {}".format(image_disk))
+                image_disk = os.path.join(gluster_mount_dir, image_disk)
+            cmd = "stat {:s}".format(image_disk)
+            pattern = r"Blocks:\s(\d+)"
+            try:
+                block_num = re.search(pattern, process.system_output(cmd)).groups()[0]
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: pls add the step description

File: qemu/tests/thin_provisioning_image.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                time.sleep(wait_time)
+                block_num_after_sec = re.search(pattern, process.system_output(cmd)).groups()[0]
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: pls add the step description

File: qemu/tests/thin_provisioning_image.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+                logging.info(
+                    'Disk is not steady yet, block_num: {:s}, block_num_after_{}sec:{:s}'.format(
+                        block_num, wait_time, block_num_after_sec))
+                if block_num == block_num_after_sec:
+                    return block_num
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: what need to do if block_num != block_num_after_sec

File: qemu/tests/thin_provisioning_image.py [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+    if enable_gluster == "yes":
+        gluster_volume_mount = gluster.create_gluster_uri(params)
+        gluster_volume_mount = gluster_volume_mount.replace(":0", ":")
+        gluster_volume_mount = gluster_volume_mount.replace("gluster://", "")
+        gluster_volume_mount = gluster_volume_mount.rstrip('/')
+        tempdir = auto_data_dir.get_tmp_dir()
+        utils_misc.mount(gluster_volume_mount, tempdir, 'glusterfs')
+
+        params["only_disk_enable_gluster"] = "yes"
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: why not set in cfg file?

File: qemu/tests/thin_provisioning_image.py [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+            params, params["image_name_gluster_disk"], params["image_format_gluster_disk"]))
+        params["image_name_gluster_disk"] = image_file_name
+        cmd = "qemu-img create -f {} {} {}".format(params["image_format_gluster_disk"],
+                                                   image_file_name, params["image_size_gluster_disk"])
+        try:
+            process.system_output(cmd)
+        except:
+            raise error.TestError("Failed to create disk image in gluster server")
+    else:
+        params["image_name_local_disk"] = "local_disk"
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: pls make it configurable

File: qemu/tests/thin_provisioning_image.py [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+    session.cmd("rm -rf %s" % ddfile)
+    time.sleep(5)
+    error.context("TEST STEP 6: Execute fstrim in guest", logging.info)
+    fstrim_cmd = params["fstrim_cmd"]
+    session.cmd(fstrim_cmd, timeout=timeout)
+    block_num_after_trim = count_block_num(image_file_name, tempdir)
+    error.context("block_num_after_trim: {:s}".format(block_num_after_trim), logging.info)
+    if params.get("image_format_local_disk") == "qcow2" or params.get("image_format_gluster_disk") == "qcow2":
+        if params.get("guest_filesystem") == "ext4":
+            if block_num_after_trim != "277520":
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: what does these nums mean

Comment (PyLearner) [inline qemu/tests/thin_provisioning_image.py]: test it manually , will get the same num, so use it to judge.

File: qemu/tests/thin_provisioning_image.py (hunk 6 above)
Comment (PyLearner) [inline qemu/tests/thin_provisioning_image.py]: the infinite loop will run until block_num == block_num_after_sec

File: qemu/tests/thin_provisioning_image.py [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+    timeout = float(params.get("login_timeout", 240))
+    session = vm.wait_for_login(timeout=timeout)
+
+    guest_device_name = thin_provisioning.get_guest_discard_disk(session)
+    format_all_disk_cmd = params["format_all_disk_cmd"]
+    format_all_disk_cmd = format_all_disk_cmd.replace("DISK1", guest_device_name)
+    error.context("TEST STEP 3: Format disk '{:s}' in guest".format(guest_device_name), logging.info)
+    session.cmd(format_all_disk_cmd)
+    block_num_after_format = count_block_num(image_file_name, tempdir)
+    error.context("block_num_after_format: {:s}".format(block_num_after_format), logging.info)
```
Comment (suqinhuang) [inline qemu/tests/thin_provisioning_image.py]: it's better to use error_context.context

**Comment by xutian** [comment]
@PyLearner  there some comments from @suqinhuang  still not get reponse, please update your pull request if no any explain, else reply the comment

**Comment by PyLearner** [comment]
@xutian, thanks for the review ,i will update .

**Pull request closed without merging**